widget/embedded/src/hooks/useStatefulConnect/useStatefulConnect.ts

```diff
@@ -0,0 +1,222 @@
+
+    if (isDisconnected) {
+      if (!wallet.namespaces) {
+        return await runConnect(wallet.type, undefined, options);
+      }
+
+      const needsNamespace = wallet.namespaces.length > 1;
+      const needsDerivationPath = wallet.needsDerivationPath;
+
+      if (needsNamespace) {
+        dispatch({
+          type: 'needsNamespace',
+          payload: {
+            providerType: wallet.type,
+            providerImage: wallet.image,
+            availableNamespaces: wallet.namespaces,
+            singleNamespace: wallet.singleNamespace,
+          },
+        });
+        return { status: ResultStatus.Namespace };
+      } else if (needsDerivationPath) {
+        dispatch({
+          type: 'needsDerivationPath',
+          payload: {
+            providerType: wallet.type,
+            providerImage: wallet.image,
+            namespace: wallet.namespaces[0],
+          },
+        });
+        return { status: ResultStatus.DerivationPath };
+      }
+
+      return await runConnect(
+        wallet.type,
+        wallet.namespaces.map((namespace) => ({ namespace })),
+        options
+      );
+    }
+
+    if (options?.disconnectIfConnected) {
+      await handleDisconnect(wallet.type);
+      return { status: ResultStatus.Disconnected };
+    }
+
+    return { status: ResultStatus.DisconnectedUnhandled };
+  };
+
+  const handleNamespace = async (
+    wallet: WalletInfoWithExtra,
+    selectedNamespaces: Namespace[]
+  ): Promise<Result> => {
+    const isSingleNamespace = wallet.singleNamespace;
+    const needsDerivationPath = wallet.needsDerivationPath;
+    const firstSelectedNamespace = selectedNamespaces[0];
+
+    if (!firstSelectedNamespace) {
+      throw new Error(
+        'To confirm a namespace, you should select at least one namespace.'
+      );
+    }
+
+    /*
+     * Currently we support derivation path only for single namespace wallets.
+     *
+     * Basically Single namespace is a list with radio buttons to let the user select one value within the list.
+     */
+    if (isSingleNamespace && needsDerivationPath) {
+      dispatch({
+        type: 'needsDerivationPath',
+        payload: {
+          providerType: wallet.type,
+          providerImage: wallet.image,
+          namespace: firstSelectedNamespace,
+        },
+      });
+
+      return { status: ResultStatus.DerivationPath };
+    }
+
+    /**
+     * If execution reaches here, means we don't have any more optional step (like namespace and derivation)
+     * So we should start connecting wallet.
+     */
+    const type = connectState.namespace!.providerType;
+    const namespaces = selectedNamespaces.map((namespace) => ({
+      namespace,
+    }));
+
+    return await runConnect(type, namespaces);
+  };
+
+  const handleDerivationPath = async (
+    derivationPath: string
+  ): Promise<Result> => {
+    if (!derivationPath) {
+      throw new Error(
+        "Derivation path is empty. Please make sure you've filled the field correctly."
```

Comment:
```suggestion
        "Derivation Path can not be empty."
```

Reply:
Original text has a suggested action next to the error. I prefer to keep the original text.